modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/CreditRatesProvider.java

```diff
@@ -0,0 +1,649 @@
+   * <p>
+   * The curve data, predicting the discount factor, associated with each currency.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<Currency, CreditDiscountFactors> discountCurves;
+  /**
+   * The credit rate curves.
+   * <p>
+   * The curve date, predicting the recovery rate, associated with each legal entity.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<StandardId, RecoveryRates> recoveryRateCurves;
+
+  //-------------------------------------------------------------------------
+  @ImmutablePreBuild
+  private static void preBuild(Builder builder) {
+    if (builder.valuationDate == null && !builder.discountCurves.isEmpty()) {
+      builder.valuationDate = builder.discountCurves.values().iterator().next().getValuationDate();
+    }
+  }
+
+  @ImmutableValidator
+  private void validate() {
+    for (Entry<Pair<StandardId, Currency>, CreditDiscountFactors> entry : creditCurves.entrySet()) {
+      if (!entry.getValue().getValuationDate().isEqual(valuationDate)) {
+        throw new IllegalArgumentException("Invalid valuation date for the credit curve: " + entry.getValue());
+      }
+    }
+    for (Entry<Currency, CreditDiscountFactors> entry : discountCurves.entrySet()) {
+      if (!entry.getValue().getValuationDate().isEqual(valuationDate)) {
+        throw new IllegalArgumentException("Invalid valuation date for the discount curve: " + entry.getValue());
+      }
+    }
+    for (Entry<StandardId, RecoveryRates> entry : recoveryRateCurves.entrySet()) {
+      if (!entry.getValue().getValuationDate().isEqual(valuationDate)) {
+        throw new IllegalArgumentException("Invalid valuation date for the recovery rate curve: " + entry.getValue());
+      }
+    }
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the survival probabilities for a standard ID and a currency.
+   * <p>
+   * If both the standard ID and currency are matched, the relevant {@code LegalEntitySurvivalProbabilities} is returned. 
+   * <p>
+   * If the valuation date is on the specified date, the survival probability is 1.
+   * 
+   * @param legalEntityId  the standard ID of legal entity to get the discount factors for
+   * @param currency  the currency to get the discount factors for
+   * @return the survival probabilities 
+   * @throws IllegalArgumentException if the survival probabilities are not available
+   */
+  public LegalEntitySurvivalProbabilities survivalProbabilities(StandardId legalEntityId, Currency currency) {
+    CreditDiscountFactors survivalProbabilities = creditCurves.get(Pair.of(legalEntityId, currency));
+    if (survivalProbabilities == null) {
+      throw new IllegalArgumentException("Unable to find credit curve: " + legalEntityId + ", " + currency);
+    }
+    return LegalEntitySurvivalProbabilities.of(survivalProbabilities, legalEntityId);
```

Comment:
Why does the map hold `CreditDiscountFactors` and not `LegalEntitySurvivalProbabilities`?